main.go

```diff
@@ -135,51 +143,62 @@ func createBook(w http.ResponseWriter, r *http.Request) {
 	var newBook Book
 	err := json.NewDecoder(r.Body).Decode(&newBook) //Read the Json body and save the entry to newBook
 	if err != nil {
-		log.Println("error while decoding the json entry:", err)
-		w.WriteHeader(http.StatusBadRequest)
+		log.Println(err)
+		w.WriteHeader(http.StatusBadRequest) //SHOULDN'T IT BE STATUS INTERNAL SERVER ERROR?
 		return
 	}
 
-	blankFields := filledFields(newBook) //Verify if all entry fields are filled.
-	if blankFields != "" {
+	filled := filledFields(newBook) //Verify if all entry fields are filled.
+	if !filled {
+		error100, err := json.Marshal(errorResponse{100, "All the fields - name, price and inventory - must be filled correctly."})
+		if err != nil {
+			log.Println(err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
 		w.Header().Set("content-type", "application/json")
 		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte(blankFields))
+		w.Write(error100)
 		return
 	}
 
 	unique, returnedBook := sameNameOnDB(newBook) //Verify if the already there is a book with the same name in the database
 	if !unique {
 		sameNameBook, err := json.Marshal(returnedBook)
 		if err != nil {
-			log.Println("error while Marshalling returnedBook:", err)
+			log.Println(err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		error101, err := json.Marshal(errorResponse{101, "There is already a book with this name on database:"})
+		if err != nil {
+			log.Println(err)
 			w.WriteHeader(http.StatusInternalServerError)
 			return
 		}
 		w.Header().Set("content-type", "application/json")
 		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte(string(sameNameBook)))
+		w.Write(append(error101, sameNameBook...))
```

Comment:
You cannot just append a JSON into another, that's not a valid JSON.

Reply:
Lets return the error response only, without the duplicated book. We can discuss how to improve the error message in a call.

Reply:
Ok. I left some commented code where the duplicated book is shown until we can talk.